torchdata/dataloader2/reading_service.py

```diff
@@ -301,26 +312,31 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
         self._datapipe = datapipe
         return datapipe
 
-    def initialize_iteration(self) -> None:
+    def initialize_iteration(self, seed_generator: SeedGenerator) -> None:
         r"""
         Shares the same seed from rank 0 to other ranks across the distributed processes
         and apply the random seed to the graph of ``DataPipe``.
         """
-        # TODO: Seed Generator should be moved to DataLoader2 after the API
-        #       change of initialize_iteration is landed.
-        seed = self._share_seed()
-        _seed_generator = torch.Generator()
-        _seed_generator.manual_seed(seed)
+        shared_seed = self._share_seed(seed_generator.generate())
+        seed_generator.seed(shared_seed)
```

Comment:
We will broadcast the generated `shared_seed` on Rank 0 to other ranks. And, then we use this synchronized seed to re-seed DataLoader2 across all distributed processes.

Reply:
I see. nit: Should we rename `rng` to `dp_graph_rng` here to specify that it is specific for the DataPipe graph? It would be useful to distinguish `seed_generator` (which is for DL2) from `rng` (which is for the DP graph within DL2).

Reply:
I can rename it for now. And, I do have another follow-up PRs to change the behavior of `apply_random_seed` that would take `seed_generator`. I will write more detail reason in that PR.